editor/templates/editoritem/preview.html

```diff
@@ -54,19 +58,48 @@ <h1 class="description">
 
 
         <script>
-var share_button = document.getElementById('share');
-var embed_display = document.getElementById('embed-url');
-share_button.addEventListener('click',function() {
-    embed_display.classList.toggle('shown');
-    embed_display.focus();
-});
+
+            var share_button = document.getElementById('share');
+            var embed_display = document.getElementById('embed-url');
+            var copy_text = document.getElementById('copy-text');
+            var close_button = document.getElementById('close-button');
+
+            /* To toggle the embedded URL area */
+            share_button.addEventListener('click',function() {
+                embed_display.classList.toggle('shown'); 
+                embed_display.focus();
+            });
+
+            /* To close the embedded URL when the close is clicked */
+            close_button.addEventListener('click',function() {
+                embed_display.classList.toggle('shown'); 
+                embed_display.focus();
+            });
+
+            /* Function to copy the URL to the clipboard */
+            copy_text.addEventListener('click',function() {
+                                
+                /* To copy the URL to clipboard */
+                navigator.clipboard.writeText(`{{SITE_ROOT}}{{embed_url}}`);
+                
+                /* Displays the message that URL is copied to clipboard */
+                document.getElementById("copied-text").innerHTML = '<span class="left-right-padding">&nbsp URL Copied &nbsp</span>';
```

Comment:
Why write HTML in JS, when you could have it already in the page and toggle its visibility with CSS?